src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/rpa/MoreTimeRequestedNotificationServiceTest.java

```diff
@@ -0,0 +1,84 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import uk.gov.hmcts.cmc.claimstore.MockSpringTest;
+import uk.gov.hmcts.cmc.claimstore.events.response.MoreTimeRequestedEvent;
+import uk.gov.hmcts.cmc.claimstore.rpa.config.EmailProperties;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim;
+import uk.gov.hmcts.cmc.email.EmailAttachment;
+import uk.gov.hmcts.cmc.email.EmailData;
+
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.cmc.claimstore.rpa.ClaimIssuedNotificationService.JSON_EXTENSION;
+import static uk.gov.hmcts.cmc.claimstore.utils.DocumentNameUtils.buildJsonMoreTimeRequestedFileBaseName;
+
+public class MoreTimeRequestedNotificationServiceTest extends MockSpringTest {
+
+    @Autowired
+    private MoreTimeRequestedNotificationService service;
+    @Autowired
+    private EmailProperties emailProperties;
+
+    @Captor
+    private ArgumentCaptor<String> senderArgument;
+
+    @Captor
+    private ArgumentCaptor<EmailData> emailDataArgument;
+
+    private Claim claim;
+
+    private MoreTimeRequestedEvent event;
+
+    @Before
+    public void setUp() {
+        claim = SampleClaim
+            .builder()
+            .withRespondedAt(LocalDateTime.of(2018, 4, 26, 1, 1))
+            .build();
+
+        event = new MoreTimeRequestedEvent(claim, LocalDateTime.now().toLocalDate(), "<any-email>");
+
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerWhenGivenNullClaim() {
+        service.notifyRobotics(null);
+    }
+
+    @Test
+    public void shouldSendEmailWithConfiguredValues() {
+
+        service.notifyRobotics(event);
+
+        verify(emailService).sendEmail(senderArgument.capture(), emailDataArgument.capture());
+
+        assertThat(senderArgument.getValue()).isEqualTo(emailProperties.getSender());
+        assertThat(emailDataArgument.getValue().getTo()).isEqualTo(emailProperties.getResponseRecipient());
+        assertThat(emailDataArgument.getValue().getSubject()).isEqualToIgnoringNewLines("J additional time 000CM001");
+        assertThat(emailDataArgument.getValue().getMessage()).isEmpty();
+    }
+
+    @Test
+    public void shouldSendEmailWithConfiguredValuesAndAttachments() {
+
+        service.notifyRobotics(event);
+
+        verify(emailService).sendEmail(senderArgument.capture(), emailDataArgument.capture());
+
+        EmailAttachment moreTimeReqJsonAttachment = emailDataArgument.getValue().getAttachments().get(0);
+
+        String expectedJson = buildJsonMoreTimeRequestedFileBaseName(claim.getReferenceNumber()) + JSON_EXTENSION;
+
+        assertThat(moreTimeReqJsonAttachment.getContentType()).isEqualTo(MediaType.APPLICATION_JSON_VALUE);
+        assertThat(moreTimeReqJsonAttachment.getFilename()).isEqualTo(expectedJson);
```

Comment:
these assertions can be added to the previous test, we won't require to write a separate test for same results of a method call

Reply:
It's deliberate separation of not doing too much in one test. One is meant for all values without attachments and second one is with attachment only.

Reply:
It was deliberate action of not putting too much into one test and kept it separate purposefully.